### DISCUSSION TITLE
Can't get the basic echo example to work properly

### DISCUSSION THREAD

**Discussion by olivierpichon** [ionic-team/capacitor#3893] [Q&A] [answered]
Hi there,

I am trying to create my own plugin following the documentation.

I launch the below code in a iOS simulator:

```
  MyPlugin.echo({value: 'test'}).then(response => {
    Modals.confirm({
      title: 'Title',
      message: response.value
    })
  })
```

This shows a Modal with `test` as a message as anticipated.

Now I change slightly the `echo` implementation in Xcode to be hardcoded:

```
    @objc func echo(_ call: CAPPluginCall) {
        call.resolve([
            "value": "hardcoded"
        ])
    }
```

But this still shows the `test` message in the modal :( 
It looks like I am never actually calling the native code but rather always the web implementation. I don't understand as `Capacitor.isNative` returns true and I followed the step by step creation of a plugin without any customisation. Any advises?

**Comment by pwespi** [comment, marked as answer]
Make sure to import the plugin with:

```typescript
import { Plugins } from '@capacitor/core';
const { MyPlugin } = Plugins;
```

If you import `MyPlugin` directly, it will always use the web implementation. (This will be improved in capacitor v3 I think.)

**Comment by olivierpichon** [comment]
@pwespi I spent so much time on this, I can't thank you enough! This is now working!!
